src/subscription_manager/cli_command/status.py

```diff
@@ -46,38 +47,59 @@ def __init__(self):
             ),
         )
 
-    def _do_command(self):
-        # list status and all reasons it is not valid
+    def _get_date_cli_option(self):
+        """
+        Try to get and validate command line options date
+        :return: Return date or None, when date was not provided
+        """
         on_date = None
         if self.options.on_date:
             try:
                 on_date = entitlement.EntitlementService.parse_date(self.options.on_date)
             except ValueError as err:
                 system_exit(os.EX_DATAERR, err)
+        return on_date
+
+    def _print_status(self, service_status):
+        """
+        Print only status
+        :return: Print overall status
+        """
 
         print("+-------------------------------------------+")
         print("   " + _("System Status Details"))
         print("+-------------------------------------------+")
 
-        service_status = entitlement.EntitlementService(None).get_status(on_date)
-        reasons = service_status["reasons"]
-
-        if service_status["valid"]:
-            result = 0
-        else:
-            result = 1
-
         ca_message = ""
         has_cert = _(
             "Content Access Mode is set to Simple Content Access. This host has access to content, regardless of subscription status.\n"
         )
 
         certs = self.entitlement_dir.list_with_content_access()
-        ca_certs = [cert for cert in certs if cert.entitlement_type == CONTENT_ACCESS_CERT_TYPE]
-        if ca_certs:
-            ca_message = has_cert
+        sca_certs = [cert for cert in certs if cert.entitlement_type == CONTENT_ACCESS_CERT_TYPE]
+        sca_mode_detected = False
+
+        refresh_service = Refresh(cp=self.cp, ent_cert_lib=self.entcertlib)
+
+        if sca_certs:
+            sca_mode_detected = True
         else:
+            # When there are no entitlement SCA certificates, but status_id is "disabled", then
+            # it means that content access mode has changed on the server and entitlement certificates
+            # have to be refreshed
+            if service_status["status_id"] == "disabled":
+                refresh_service.refresh()
             if is_simple_content_access(uep=self.cp, identity=self.identity):
+                sca_mode_detected = True
+
+        if sca_mode_detected is True:
+            # When SCA mode was detected using cache or installed SCA entitlement certificates, but status_id
+            # is not "disabled", then it means that content access mode has changed on the server and entitlement
+            # certificates have to be refreshed
+            if service_status["status_id"] != "disabled":
+                log.debug("Found SCA cert, but status ID is not 'disabled'. Refreshing entitlement certs...")
```

Comment:
maybe extend this debug like eg like `... not 'disabled' ({service_status["status_id"]}). Refreshing ...`; this way we can know what's the actual status when reading this log entry

Reply:
Done